File: go/models/return_code_result.go
package models

// ReturnCodeResult represents the response from saveReturnShipmentCode.
type ReturnCodeResult struct {
	OutFlag   int    `xml:"outFlag"`
	OutResult string `xml:"outResult"`
	ErrCode   int    `xml:"errCode"`
}

// IsSuccess returns true if the operation was successful.
func (r *ReturnCodeResult) IsSuccess() bool {
	return r.OutFlag == 0
}

File: java/src/main/java/com/yurticikargo/models/ReturnCodeResult.java
package com.yurticikargo.models;

/**
 * saveReturnShipmentCode SOAP yanıt modeli.
 */
public class ReturnCodeResult {

    private int outFlag;
    private String outResult;
    private int errCode;

    public int getOutFlag() { return outFlag; }
    public void setOutFlag(int outFlag) { this.outFlag = outFlag; }

    public String getOutResult() { return outResult; }
    public void setOutResult(String outResult) { this.outResult = outResult; }

    public int getErrCode() { return errCode; }
    public void setErrCode(int errCode) { this.errCode = errCode; }

    public boolean isSuccess() { return outFlag == 0 && errCode == 0; }

    @Override
    public String toString() {
        return "ReturnCodeResult{outFlag=" + outFlag + ", outResult='" + outResult +
               "', errCode=" + errCode + "}";
    }
}
